precompiles/ArbAggregator.go

```diff
@@ -20,87 +16,111 @@ type ArbAggregator struct {
 	Address addr // 0x6d
 }
 
-// Gets an account's preferred aggregator
+var ErrNotOwner = errors.New("must be called by chain owner")
+
+// [Deprecated]
 func (con ArbAggregator) GetPreferredAggregator(c ctx, evm mech, address addr) (prefAgg addr, isDefault bool, err error) {
-	l1p := c.State.L1PricingState()
-	maybePrefAgg, err := l1p.UserSpecifiedAggregator(address)
+	posters, err := c.State.L1PricingState().BatchPosterTable().AllPosters()
 	if err != nil {
 		return common.Address{}, false, err
 	}
-	if maybePrefAgg != nil {
-		return *maybePrefAgg, false, nil
-	}
-	maybeReimbursableAgg, err := l1p.ReimbursableAggregatorForSender(address)
-	if err != nil || maybeReimbursableAgg == nil {
-		return common.Address{}, false, err
+	if len(posters) == 0 {
+		return common.Address{}, false, errors.New("no batch posters exist")
 	}
-	return *maybeReimbursableAgg, true, nil
+	return posters[0], true, err
 }
 
-// Sets the caller's preferred aggregator to that provided
-func (con ArbAggregator) SetPreferredAggregator(c ctx, evm mech, prefAgg addr) error {
-	var maybePrefAgg *common.Address
-	if prefAgg != (common.Address{}) {
-		maybePrefAgg = &prefAgg
+// [Deprecated]
+func (con ArbAggregator) GetDefaultAggregator(c ctx, evm mech) (addr, error) {
+	posters, err := c.State.L1PricingState().BatchPosterTable().AllPosters()
+	if err != nil {
+		return common.Address{}, err
+	}
+	if len(posters) == 0 {
+		return common.Address{}, errors.New("no batch posters exist")
 	}
-	return c.State.L1PricingState().SetUserSpecifiedAggregator(c.caller, maybePrefAgg)
+	return posters[0], err
 }
 
-// Gets the chain's default aggregator
-func (con ArbAggregator) GetDefaultAggregator(c ctx, evm mech) (addr, error) {
-	return c.State.L1PricingState().DefaultAggregator()
+// Get the addresses of all current batch posters
+func (con ArbAggregator) GetBatchPosters(c ctx, evm mech) ([]addr, error) {
+	return c.State.L1PricingState().BatchPosterTable().AllPosters()
 }
 
-// Sets the chain's default aggregator (caller must be the current default aggregator, its fee collector, or an owner)
+// [Deprecated]
 func (con ArbAggregator) SetDefaultAggregator(c ctx, evm mech, newDefault addr) error {
```

Comment:
As this method was privileged, can't we just remove it?

Reply:
Removed it